### DISCUSSION TITLE
STATS AND LOCALS TO IMGUI::TEXT?

### DISCUSSION THREAD

**Discussion by Jaksor23** [YimMenu/YimMenu#2222] [Q&A]
Recently I wanted to create a text that would allow me to see the code of the safe in the mod menu, but when I open it, it crashes, with no apparent error.

Code of the locals:
```
auto safe_box_stash = gta_util::find_script_thread(RAGE_JOAAT("fm_content_stash_house"));
int code_safe = *script_local(safe_box_stash, local1 + local2 + local3).as<int*>();
 
ImGui::Text("Safe Code: %d", code_safe);
```

-------------------------------------------------------------------------------------------------------------

The stat text works and updates itself, but it generates me native invoker error spam.

Code of the stat:
```
static int i    = 0;
int stat = STATS::STAT_GET_INT(RAGE_JOAAT("MPPLY_KILLS_PLAYERS"), &i, -1);
ImGui::Text("Players Killed: %d", i);
```

Surely it is skill issues or dumb question, anyway thanks in advance to the community.
![2](https://github.com/YimMenu/YimMenu/assets/129014475/b7d6fa98-a438-491b-a333-c34bf9c3f2d3)

**Comment by FIying-Scotsman** [comment]
Are you running this through the fiber pool? 

From experience calling natives outside the fiber pool will cause the menu to try and run the natives via the DX thread - which doesn’t know what the hell a native is (bar a handful which are just class wrappers like GET_PLAYER_NAME which just return rlPlayerinfo -> CName) so will either crash or just won’t work. 

That looks to be the case here.

**Comment by pierrelasse** [comment]
You will have to save this somewhere because you can't call natives in the render loop.
You must do this in the game loop.
Here is how you can do it:
```lua
local me_cheater = false

-- Game loop
script.register_looped("update", function()
  me_cheater = stats.get_bool("MPPLY_IS_CHEATER")
end)

-- Render loop
gui.get_tab("CheaterCheck"):add_imgui(function()
  ImGui.Text("R* sus?: "..tostring(me_cheater))
end)
```

**Reply by lonelybud** [reply]
this is lua. The question is asked in c++

**Reply by pierrelasse** [reply]
So he is making a custom build? o.O
Still he can get the solution from it
